clients/src/main/java/org/apache/kafka/clients/consumer/internals/MembershipManagerImpl.java

```diff
@@ -685,13 +690,6 @@ public boolean shouldHeartbeatNow() {
     @Override
     public void onHeartbeatRequestSent() {
         MemberState state = state();
-        if (isStaled()) {
-            log.debug("Member {} is staled and is therefore leaving the group.  It will rejoin upon the next poll.", memberEpoch);
-            // TODO: Integrate partition revocation/loss callback
-            transitionToJoining();
-            return;
-        }
-
```

Comment:
For the uninitiated (like myself), would you consider adding a really brief comment that provides pointers to where some of the other states (e.g. `STALE`) are handled?

Reply:
Do you mean comment here? The handling is all over the manager really, but the main logic is in the transition functions I would say: `transitionToJoining`, `transitionToFenced`, `transitionToFatal`, `transitionToSendingLeaveGroup`, `transitionToUnsubscribed`, `transitionToStale`, and the `reconcile` of course. Each state handling func usually needs to take specific actions for the transitions, but also considering how things might have changed. This is a state machine that changes with inputs from 2 sides: API calls, and broker HB responses, so lots of things can change, anytime, while we transition from A to B.